### ISSUE TITLE
offline install krew

### ISSUE THREAD

**Issue by willzhang** [kubernetes-sigs/krew#759]
github is blocked by china ovenment, so anyway offline install krew?

```
[root@harbor ~]# (
>   set -x; cd "$(mktemp -d)" &&
>   OS="$(uname | tr '[:upper:]' '[:lower:]')" &&
>   ARCH="$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/')" &&
>   KREW="krew-${OS}_${ARCH}" &&
>   curl -fsSLO "https://github.com/kubernetes-sigs/krew/releases/latest/download/${KREW}.tar.gz" &&
>   tar zxvf "${KREW}.tar.gz" &&
>   ./"${KREW}" install krew
> )
++ mktemp -d
+ cd /tmp/tmp.dFvpYdM5Xb
++ uname
++ tr '[:upper:]' '[:lower:]'
+ OS=linux
++ uname -m
++ sed -e s/x86_64/amd64/ -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/'
+ ARCH=amd64
+ KREW=krew-linux_amd64
+ curl -fsSLO https://github.com/kubernetes-sigs/krew/releases/latest/download/krew-linux_amd64.tar.gz
curl: (35) TCP connection reset by peer
```

```
[root@harbor ~]# wget https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.tar.gz
--2022-03-13 08:44:07--  https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.tar.gz
Resolving github.com (github.com)... 20.205.243.166
Connecting to github.com (github.com)|20.205.243.166|:443... failed: Connection refused.
```

download package still have probles
```
[root@harbor ~]# ./krew-linux_amd64 install krew
Adding "default" plugin index from https://github.com/kubernetes-sigs/krew-index.git.
W0313 08:48:52.286208   28616 update.go:168] failed to update index "default": fetch index at "/root/.krew/index/default" failed: command execution failure, output="fatal: unable to access 'https://github.com/kubernetes-sigs/krew-index.git/': TCP connection reset by peer\n": exit status 128
failed to update the following indexes: default
: fetch index at "/root/.krew/index/default" failed: command execution failure, output="fatal: unable to access 'https://github.com/kubernetes-sigs/krew-index.git/': TCP connection reset by peer\n": exit status 128
```

what i must do , why not put krew.yaml in krew.tar.gz?
```
wget https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.tar.gz
wget https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.yaml
tar -zxvf krew.tar.gz
./krew-linux_amd64 install --manifest=krew.yaml --archive=krew-linux_amd64.tar.gz
```

**Comment by willzhang** [comment]
it's so hard access github
```
[root@Ops-repo ~]# kubectl krew install rook-ceph
Adding "default" plugin index from https://github.com/kubernetes-sigs/krew-index.git.
failed to add default plugin index in absence of no indexes: command execution failure, output="Cloning into '/root/.krew/index/default'...\nfatal: unable to access 'https://github.com/kubernetes-sigs/krew-index.git/': Network file descriptor is not connected\n": exit status 128
[root@Ops-repo ~]# 
[root@Ops-repo ~]# 
[root@Ops-repo ~]# 
```

**Comment by ahmetb** [comment]
This is not a use case we thought about. How do other projects provide their binaries to CN?

**Comment by ahmetb** [comment]
Thinking more about this, even if we made it somehow easier to install Krew itself, all the plugin binaries you'd get are still going to be downloaded from GitHub and have the same fate.

**Comment by yxxhero** [comment]
@willzhang try use http proxy.

**Comment by willzhang** [comment]
> Thinking more about this, even if we made it somehow easier to install Krew itself, all the plugin binaries you'd get are still going to be downloaded from GitHub and have the same fate.

I have other ways to download binary files. I just need to know where to download them and install to where after downloading

**Comment by ahmetb** [comment]
It doesn't make a lot of sense to me to use a package manager if you'll do all its job manually. 

Nearly all kubectl plugins have a manual installation guide(without Krew) on their READMEs, so you can follow it. Most of them are as easy as putting the executable anywhere in your PATH.

Closing this request as I don't think we can do anything to make your life easier. Feel free to open a new issue if you're failing to use a proxy/VPN with Krew. 

/close

**Comment by snail2sky** [comment]
Due to network restrictions, Chinese users can download the krew binary package through other means, and can manually transfer the krew binary package to the server. The server cannot connect to github. However, it can provide the server with a proxy site for github, such as github-proxy.com. Is there a place for krew to configure the configuration of github global proxy?
